helm/cluster-azure/values.schema.json

```diff
@@ -197,8 +197,29 @@
                     "title": "Identity",
                     "type": "object"
                 },
+                "image": {
+                    "properties": {
+                        "gallery": {
+                            "default": "gsCAPITest1-5cb24dcf-a2d0-4aba-820f-b52ca78f96e6",
+                            "title": "Name of the Community gallery hosting the image",
+                            "type": "string"
+                        },
+                        "name": {
+                            "default": "{{ printf \"capi-flatcar-stable-%s-gen2\" .Values.internal.kubernetesVersion }}",
```

Comment:
⚠️ I know this is /internal, but anyway, this is really brittle. Can the logic be done in the actual template? Maybe you can ask only for the printf format argument here?

```suggestion
                            "default": "capi-flatcar-stable-%s-gen2",
```

Reply:
I know it is ugly, I can try something else :+1:

Reply:
I don't think we can do it like that , because if we move the `helm function` out of the variable name we risk 

```
capi-flatcar-stable-XXX-gen2%!(EXTRA string=1.24.11)
```

if a value without `-%s` is passed in

Reply:
@marians please check updated code